routers/web/repo/view.go

```diff
@@ -839,24 +840,29 @@ func renderDirectoryFiles(ctx *context.Context, timeout time.Duration) git.Entri
 		return nil
 	}
 
+	// the code below heavily depends on latestCommit and it doesn't
+	// make much sense to do a incomplete directory rendering if it's nil
+	if latestCommit == nil {
+		ctx.ServerError("GetCommitsInfo", errors.New("latestCommit is nil"))
```

Comment:
No this is not an error. If it is nil, it has just taken longer than expected to resolve the latest commit.

We need to tolerate latest commit being nil instead of throwing an error.